lib/events/stream.go

```diff
@@ -655,7 +655,7 @@ func (w *sliceWriter) completeStream() {
 		case upload := <-w.completedUploadsC:
 			part, err := upload.getPart()
 			if err != nil {
-				log.WithError(err).Warningf("Failed to upload part.")
+				log.WithError(err).Warningf("Failed to upload %q part for session id %q", w.proto.cfg.Upload.ID, w.proto.cfg.Upload.SessionID)
```

Comment:
This sounds like a part ID now though. I wonder if this would be better.
```suggestion
				log.WithError(err).
					WithField("upload", w.proto.cfg.Upload.ID).
					WithField("session", w.proto.cfg.Upload.SessionID). 
					Warning("Failed to upload part")
```